lntest/itest/signer_test.go

```diff
@@ -0,0 +1,196 @@
+				"failed to derive the expected key",
+			)
+		}
+	}
+
+	nodePub, err := btcec.ParsePubKey(net.Alice.PubKey[:], btcec.S256())
+	require.NoError(t.t, err, "failed to parse node pubkey")
+
+	customizedKeyFamily := int32(keychain.KeyFamilyMultiSig)
+	customizedIndex := int32(1)
+	customizedPub, err := deriveCustomizedKey(
+		ctxb, net.Alice, customizedKeyFamily, customizedIndex,
+	)
+	require.NoError(t.t, err, "failed to create customized pubkey")
+
+	// Test DeriveSharedKey with no optional arguments. It will result in
+	// performing an ECDH between the ephemeral key and the node's pubkey.
+	req := &signrpc.SharedKeyRequest{EphemeralPubkey: ephemeralPubBytes}
+	assertECDHMatch(nodePub, req)
+
+	// Test DeriveSharedKey with a KeyLoc which points to the node's pubkey.
+	req = &signrpc.SharedKeyRequest{
+		EphemeralPubkey: ephemeralPubBytes,
+		KeyLoc: &signrpc.KeyLocator{
+			KeyFamily: int32(keychain.KeyFamilyNodeKey),
+			KeyIndex:  0,
+		},
+	}
+	assertECDHMatch(nodePub, req)
+
+	// Test DeriveSharedKey with a KeyLoc being set in KeyDesc. The KeyLoc
+	// points to the node's pubkey.
+	req = &signrpc.SharedKeyRequest{
+		EphemeralPubkey: ephemeralPubBytes,
+		KeyDesc: &signrpc.KeyDescriptor{
+			KeyLoc: &signrpc.KeyLocator{
+				KeyFamily: int32(keychain.KeyFamilyNodeKey),
+				KeyIndex:  0,
+			},
+		},
+	}
+	assertECDHMatch(nodePub, req)
+
+	// Test DeriveSharedKey with RawKeyBytes set in KeyDesc. The RawKeyBytes
+	// is the node's pubkey bytes. Because the KeyLoc is nil, it will end up
+	// using the node's pubkey to perform the ECDH.
+	req = &signrpc.SharedKeyRequest{
+		EphemeralPubkey: ephemeralPubBytes,
+		KeyDesc: &signrpc.KeyDescriptor{
+			RawKeyBytes: net.Alice.PubKey[:],
+		},
+	}
+	assertECDHMatch(nodePub, req)
+
+	// Test DeriveSharedKey with a KeyLoc which points to the customized
+	// public key.
+	req = &signrpc.SharedKeyRequest{
+		EphemeralPubkey: ephemeralPubBytes,
+		KeyLoc: &signrpc.KeyLocator{
+			KeyFamily: customizedKeyFamily,
+			KeyIndex:  customizedIndex,
+		},
+	}
+	assertECDHMatch(customizedPub, req)
+
+	// Test DeriveSharedKey with a KeyLoc being set in KeyDesc. The KeyLoc
+	// points to the customized public key.
+	req = &signrpc.SharedKeyRequest{
+		EphemeralPubkey: ephemeralPubBytes,
+		KeyDesc: &signrpc.KeyDescriptor{
+			KeyLoc: &signrpc.KeyLocator{
+				KeyFamily: customizedKeyFamily,
+				KeyIndex:  customizedIndex,
+			},
+		},
+	}
+	assertECDHMatch(customizedPub, req)
+
+	// Test DeriveSharedKey with RawKeyBytes set in KeyDesc. The RawKeyBytes
+	// is the customized public key. The KeyLoc is also set with the family
+	// being the customizedKeyFamily.
+	req = &signrpc.SharedKeyRequest{
+		EphemeralPubkey: ephemeralPubBytes,
+		KeyDesc: &signrpc.KeyDescriptor{
+			RawKeyBytes: customizedPub.SerializeCompressed(),
+			KeyLoc: &signrpc.KeyLocator{
+				KeyFamily: customizedKeyFamily,
+			},
+		},
+	}
+	assertECDHMatch(customizedPub, req)
+
+	// Now we test DeriveSharedKey with invalid params.
+
```

Comment:
nit: extra newline. Maybe just merge this comment with another one?

Reply:
sure, merged.